wavefront_python_sdk/direct_ingestion/wavefront_direct_ingestion_client.py

```diff
@@ -0,0 +1,135 @@
+from __future__ import absolute_import, division, print_function
+
+from wavefront_python_sdk.common.connection_handler import IConnectionHandler
+from wavefront_python_sdk.common.utils import metric_to_line_data, \
+    histogram_to_line_data, tracing_span_to_line_data
+
+import sys
+import requests
+from threading import Timer
+from queue import Queue
+
+
+class WavefrontDirectIngestionClient(IConnectionHandler):
+    WAVEFRONT_METRIC_FORMAT = 'wavefront'
+    WAVEFRONT_HISTOGRAM_FORMAT = 'histogram'
+    WAVEFRONT_TRACING_SPAN_FORMAT = 'trace'
+
+    def __init__(self, builder):
+        IConnectionHandler.__init__(self)
+        self._default_source = "wavefrontDirectSender"
+        self._builder = builder
+        self._metrics_buffer = Queue(builder.max_queue_size)
+        self._histograms_buffer = Queue(builder.max_queue_size)
+        self._tracing_spans_buffer = Queue(builder.max_queue_size)
+        self._headers = {'Content-Type': 'text/plain', 'Authorization': 'Bearer ' + builder.token}
+        self._flush()
+
+    class Builder:
+
+        def __init__(self, server, token):
```

Comment:
I would suggest a more pythonic (and simpler) approach - avoid builder pattern altogether and simply declare optional arguments as parameters with default value:
```
    def __init__(self, server, token, max_queue_size=50000, batch_size=10000, flush_interval_seconds=1):
```

Or, if a builder pattern is preferred, use a parameterless constructor and have server and token as builder methods, and throw an exception in build() if either is null. This would be more readable too:

```
client = WavefrontDirectIngestionClient.Builder().server("https://...").token("<TOKEN>").build()
```

Reply:
Removed builder pattern for both clients: https://github.com/wavefrontHQ/wavefront-python-sdk/pull/1/commits/174878239dd09fdc23d0ac6f54090796a1d1d43b